### DISCUSSION TITLE
How search Json column as string with supabase?

### DISCUSSION THREAD

**Discussion by codestergit** [supabase/supabase#8003] [Questions]
In postgres we can search json column as text

select *
from reports
where params::text like '%abc%';

How to do this in supabase?

**Comment by steve-chavez** [comment]
Hm, you really want to search `abc` across all json keys as well? If not, you could try doing `like('param->>key', '%abc%')`([ref](https://supabase.com/docs/reference/javascript/select#querying-json-data)).

---

Otherwise:

```sql
create or replace function params_text(reports) returns text as $$
  select params::text;
$$ stable language sql;
```

```js
const { data, error } = await supabase
  .from('reports')
  .select(`*`)
  .like('params_text', '%abc%')
```

---

Casting on a column filter is not allowed out of the box through the API because clients could invalidate the indexes you already set up and produce slow queries. If you index your computed column it should be fine though(see [here](https://postgrest.org/en/stable/api.html#computed-columns)).
